cpp/src/barretenberg/honk/pcs/ipa/ipa.hpp

```diff
@@ -2,81 +2,76 @@
-        // To check poly_degree is greater than zero and a power of two
-        // TODO(#220)(Arijit): To accomodate non power of two poly_degree
+        // Checks poly_degree is greater than zero and a power of two
+        // In the future, we might want to consider if non-powers of two are needed
         ASSERT((poly_degree > 0) && (!(poly_degree & (poly_degree - 1))) &&
                "The poly_degree should be positive and a power of two");
 
         auto a_vec = polynomial;
-        // TODO(#220)(Arijit): to make it more efficient by directly using G_vector for the input points when i = 0 and
-        // write the output points to G_vec_local. Then use G_vec_local for rounds where i>0, this can be done after we
-        // use SRS instead of G_vector.
+        // TODO(#479): restructure IPA so it can be integrated with the work_queue (or replacement) and
+        // see if reducing the size of G_vec_local and b_vec by taking the first iteration out of the loop
+        // can also be integrated.
         auto srs_elements = ck->srs.get_monomial_points();
-        std::vector<CommitmentAffine> G_vec_local(poly_degree);
-        for (size_t i = 0; i < poly_degree; i++) {
-            G_vec_local[i] = srs_elements[i];
+        std::vector<Commitment> G_vec_local(poly_degree);
+        // The SRS stored in the commitment key is the result after applying the pippenger point table so the
+        // values at odd indices contain the point {srs[i-1].x * beta, srs[i-1].y}, where beta is the endomorphism
+        // G_vec_local should use only the original SRS thus we extract only the even indices.
+        for (size_t i = 0; i < poly_degree * 2; i += 2) {
+            G_vec_local[i >> 1] = srs_elements[i];
         }
-        // Construct b vector
-        // TODO(#220)(Arijit): For round i=0, b_vec can be derived in-place.
-        // This means that the size of b_vec can be 50% of the current size (i.e. we only write values to b_vec at the
-        // end of round 0)
         std::vector<Fr> b_vec(poly_degree);
         Fr b_power = 1;
         for (size_t i = 0; i < poly_degree; i++) {
             b_vec[i] = b_power;
             b_power *= opening_pair.challenge;
         }
-        // Iterate for log_2(poly_degree) rounds to compute the round commitments.
-        const size_t log_poly_degree = static_cast<size_t>(numeric::get_msb(poly_degree));
-        std::vector<Commitment> L_elements(log_poly_degree);
-        std::vector<Commitment> R_elements(log_poly_degree);
-        size_t round_size = poly_degree;
+        // Iterate for log(poly_degree) rounds to compute the round commitments.
+        auto log_poly_degree = static_cast<size_t>(numeric::get_msb(poly_degree));
+        std::vector<GroupElement> L_elements(log_poly_degree);
+        std::vector<GroupElement> R_elements(log_poly_degree);
+        std::size_t round_size = poly_degree;
 
         for (size_t i = 0; i < log_poly_degree; i++) {
             round_size >>= 1;
```

Comment:
I'd add a todo to add one iteration of this cycle as a primitive to the work queue. Seems like the most  natural way to implement Work Queue for IPA